src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,47 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter DATA_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    private static final int DATE_INDEX = 0;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate startDate = LocalDate.parse(dateFrom, DATA_FORMATTER);
+        LocalDate endDate = LocalDate.parse(dateTo,DATA_FORMATTER);
+        String[] splitData;
+        LocalDate date;
+        String dataName;
+        int hour;
+        int wage;
+        int salary;
+
+        StringBuilder sb = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+
+        for (int i = 0; i < names.length; i++) {
+            salary = 0;
+            for (String line: data) {
+                splitData = line.split(" ");
```

Comment:
All magic numbers, strings, symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)
```suggestion
                splitData = line.split(SEPARATOR);
```